### PR TITLE
Feature/starr/register user api

### PR DESCRIPTION
**Pull request by josephsPrograms**

### CODE REVIEW

**Review by campyne — APPROVED** [review]
Looks good to me. I ran it and it seemed like it was inserting into the database like it should.

File: securemailbox/views/register.py [hunk 1]
```diff
@@ -9,4 +12,43 @@
...
+
+    # Ensure all fields in request are not NULL
+    for field in ["fingerprint"]:
+        field_to_check = request.json.get(field, None)
+        if field_to_check is None:
+            return (
+                jsonify({"success": False, "error": f"field '{field}' is required."}),
+                400,
+            )
+        if len(field_to_check) > FINGERPRINT_LENGTH:
```
Comment (campyne) [inline securemailbox/views/register.py]: I think once we determine the length that fingerprints will be we should reject anything that isn't exactly that length but for now this seems fine.

File: securemailbox/constants.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Declare any variables (as constants) that are or could be used anywhere in the application
+
+# TODO: Determine what a good fingerprint length is
+FINGERPRINT_LENGTH = 100
```
Comment (samgomena) [inline securemailbox/constants.py]: Nit: Github complains with no newline at EOF.
```suggestion
FINGERPRINT_LENGTH = 100

```

**Review by samgomena — APPROVED** [review]

File: securemailbox/constants.py (hunk 2 above)
Comment (xingchengyusi) [inline securemailbox/constants.py]: Now, GPG uses 40 hex digits (160 bits SHA-1) fingerprint. 100 is too long.
The OpenPGP specification (which contains GPG) includes the [description](https://tools.ietf.org/html/rfc4880#section-12.2) of the length of a fingerprint.

**Review by xingchengyusi — APPROVED** [review]

File: securemailbox/constants.py (hunk 2 above)
Comment (josephsPrograms) [reply securemailbox/constants.py]: Oh my bad.  Idk how I missed that but I'll fix irt.

Comment (josephsPrograms) [reply securemailbox/constants.py]: Perfect thanks!

**Pull request merged**